### PR TITLE
Infra 2719 memory leak

### PR DESCRIPTION
**Pull request by julienchoiniere**
Applied 2 PR from the bosun repo:

- Simplified Redis connections + removed lots of Defer
- Couples lastInfo dropping with redis index dropping; Recommended to help scale when processing large influx of data

### CODE REVIEW

File: cmd/bosun/search/search.go [hunk 1]
```diff
@@ -91,6 +94,8 @@ func (s *Search) redisIndex(c <-chan *opentsdb.DataPoint) {
 		}
 	}
 	for dp := range c {
+		// Redis indexing does not have a dependency o lastIndex. Run in parallel
+		go s.lastIndex(dp)
```
Comment (mathpl) [inline cmd/bosun/search/search.go]: This seems risky, there's no upper limit to the amount of goroutine spawned by this. Why not simply drop `go` and index data at whatever rate we can?

Comment (julienchoiniere) [reply cmd/bosun/search/search.go]: Changes rolled back

**Comment by julienchoiniere** [comment]
Removed search backup/loading, this contained the goroutine that hang in production with multiple GB of data.
Local tests have been successful locally, instead of memory usage slowly climbing it has remained stable.

File: cmd/bosun/docker/build/Dockerfile [hunk 2]
```diff
@@ -1,30 +1,25 @@
...
 ENV HBASE /hbase/hbase-$HBASEVER
-ENV JAVA_HOME /usr/lib/jvm/java-7-openjdk-amd64
 
 RUN mkdir -p /hbase \
     && curl -SL http://archive.apache.org/dist/hbase/$HBASEVER/hbase-$HBASEVER-bin.tar.gz \
     | tar -xzC /hbase \
     && mv /hbase/hbase-$HBASEVER /hbase/hbase
 
-RUN curl -SL https://storage.googleapis.com/golang/go1.6.linux-amd64.tar.gz \
+RUN curl -SL https://storage.googleapis.com/golang/go1.7.linux-amd64.tar.gz \
```
Comment (gesanderson) [inline cmd/bosun/docker/build/Dockerfile]: Why 1.7 ? Is there something preventing us from using the latest version ?

Comment (julienchoiniere) [reply cmd/bosun/docker/build/Dockerfile]: Nothing is explicitly preventing us from upgrading the latest, the move from 1.6 to 1.7 was to benefit from the changes made to the garbage collector.
Those changes reduces GC pauses and ideally should help with our idle goroutines.

However, the patch notes for 1.12 does seem promising when it comes to memory release/allocation.
I'll test it locally and see if we can use it right away.

**Pull request merged**